### ISSUE TITLE
enableLocationRequest promise never gets resolved on Android

### ISSUE THREAD

**Issue by ickata** [NativeScript/nativescript-geolocation#91]
### Which platform(s) does your issue occur on?

- Android 7.1 Emulator

### Please, provide the following version numbers that your issue occurs with:

- CLI: 3.3.0
- Cross-platform modules: 3.2.0
- Runtime(s): 3.2.0
- Plugin(s): 

```
    "nativescript-drop-down": "3.1.1",
    "nativescript-geolocation": "4.2.0",
    "nativescript-google-maps-sdk": "2.3.2",
    "nativescript-ios-uuid": "1.0.0",
    "nativescript-iqkeyboardmanager": "1.0.1",
    "nativescript-local-notifications": "1.2.2",
    "nativescript-phone": "1.3.0",
    "nativescript-push-notifications": "0.1.5",
    "nativescript-theme-core": "1.0.4",
    "tns-core-modules": "3.2.0"
```

### Please, tell us how to recreate the issue in as much detail as possible. 

Well, simply request location permission on Android device.

### Is there any code involved? 

```
      geolocation.enableLocationRequest().then( function () {
         // I never get here...
      });
```

**Comment by ickata** [comment]
Update: It appears that location access request is not sent to the device. I came to this conclusion because I was asked for location access only the first time I opened the app after it was installed. Every next opening `geolocation.enableLocationRequest()` simply does nothing.

Also, when first asked for location, I granted access, however there is no access to location even in the device settings.

Please advise.

**Comment by DimitarTachev** [comment]
Hi @ickata,

I looked at the enableLocationRequest implementation once again and made some manual checks on Nexus 5 Android 7.1.1 emulator. During my tests, the method is always resolving or rejecting the returned promise.

Could you test the demo app from my [investigate-issue-91 branch](https://github.com/NativeScript/nativescript-geolocation/tree/tachev/investigate-issue-91) on your side and update us back if the issue is still reproducible? I'm constantly receiving 'ok' consol logs during my manual checks.

A possible reason for such behavior could be an exception in the emulator because of a missing or outdated Google Play Services (the plugin requires 11.4.0+). In order to test the geolocation plugin on an emulator, you should use emulators with Google Play Services included (as the emulators included in Android Studio 3.0).

**Comment by ickata** [comment]
Hi @DimitarTachev ,

Thank you for the response. I did first try with Google Play Services 11.2.+ (as instructed from the Readme), however the build always failed. Now I tried with the version you recommended and it fails again, with the same error:

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':processF0F1F2F3F4F5DebugGoogleServices'.
> Please fix the version conflict either by updating the version of the google-services plugin (information about the latest version is available at https://bintray.com/android/android-tools/com.google.gms.google-services/) or updating the version of com.google.android.gms to 11.4.0+.
```

**Comment by DimitarTachev** [comment]
Thanks for the details, @ickata.

After a further investigation, I was able to reproduce the issue.

The exception is caused by the push-notifications plugin. It is using an outdated version of Google Play Services and does not support modifications through a project property. There are some issues logged in the push plugin repo explaining that currently, it cannot be used together with other plugins using the play services library.

In other words, you could use the geolocation and google maps plugins together but you should remove the push one for now.

Improving the push plugin is part of our backlog and most probably will be done in a few months but I cannot give you an exact timeframe for that fix.

We will update you back here if we find a workaround for using the push plugin along with the geolocation one.

**Comment by ickata** [comment]
Oh, bad news. 

Unfortunately dropping the push-plugin is not an option. There is a reason to be there – my app relies on push notifications.

I understand that you guys have tons of tasks on the plate, however I am not sure that this issue is for the backlog. In my opinion it should be considered as high priority, as I'm sure that there are many other apps out there using the push plugin. Fixing this issue in a few months means that app updates simply won't be released, which may be too severe for any business. Can you guys please reconsider the priority? I'll be happy to help with whatever I can...

**Comment by DimitarTachev** [comment]
After a further investigation in the Push plugin, it seems like the conflict is caused by a dependency of its firebase messaging library and its version is configurable.

@ickata, try setting the following section in your app.gradle file:
```
  project.ext {
    googlePlayServicesVersion = "11.4.0"
    firebaseMessagingVersion = "11.4.0"
  }
```
using this configuration, we were able to build our project using your app dependencies.

**Comment by DimitarTachev** [comment]
@ickata, Are you using an Emulator or a real Device? If you use an emulator, you need to [send a location manually from its settings](https://screencast.com/t/I5CWrUEy2OpU).

**Comment by ickata** [comment]
Yes, emulator and yes – sending it manually. No effect...

**Comment by DimitarTachev** [comment]
Could you try requesting the location with a specified settings like the example below:
```
import { Accuracy } from "tns-core-modules/ui/enums";
...

geolocation.getCurrentLocation({
        desiredAccuracy: Accuracy.high,
        maximumAge: 5000,
        timeout: 10000
    }).then...
```

**Comment by ickata** [comment]
@DimitarTachev still no success. I can see that Google Maps detects the location when it is sent (shows the blue dot on the right place) however the geolocation plugin still does not work...

**Comment by ickata** [comment]
@DimitarTachev : `geolocation.getCurrentLocation({ timeout : 0 })` works.

**Comment by karthiksbhat** [comment]
@DimitarTachev In your comment you mentioned using Accuracy.any could cause this behavior on devices without a mobile network? Is there a reliable way to get (at least) a course location on devices without network (or on airplane mode)?

I've been having the same issue on my app, seemingly randomly, for a while now. i.e.
```
geolocation.getCurrentLocation({desiredAccuracy: Accuracy.any, timeout: 10000}).then( function ( loc ) {
      // I never get here...
   }, function ( err ) {
      // and neither here...
   });
```
Is the way to fix it `timeout: 0` or removing the timeout altogether?
